ntoskrnl/ke/amd64/mp.c

```diff
@@ -0,0 +1,95 @@
+/*
+ * PROJECT:     ReactOS Kernel
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Architecture source file to hold multiprocessor functions
+ * PROGRAMMERS:  Copyright 2021 Justin Miller <justinmiller100@gmail.com>
+ */
+
+/* INCLUDES *****************************************************************/
+
+#include <ntoskrnl.h>
+#define NDEBUG
+#include <debug.h>
+
+/* GLOBALS *******************************************************************/
+
+#define PHYSICAL_ADDRESS  LARGE_INTEGER
+PHYSICAL_ADDRESS HighestPhysicalAddress;
+ULONG ProcessorCount = 0;
+
+struct APInfo
+{
+    KPCR pcr;
+    KTSS tss;
+    ETHREAD thread;
+    KDESCRIPTOR GdtDesc;
+    KDESCRIPTOR IdtDesc;
+} APInfo;
+
+/* FUNCTIONS *****************************************************************/
+VOID
+NTAPI
+KeStartAllProcessors()
+{
+    BOOLEAN ApplicationProcessorsEnabled = FALSE;
+    HighestPhysicalAddress.QuadPart = UINT64_MAX;
+    KDESCRIPTOR GdtDesc, IdtDesc;
+    __sgdt(&GdtDesc.Limit);
+    __sidt(&IdtDesc.Limit);
+    do
+    {
+        ProcessorCount++;
+        /* Attempt to allocate memory for its new setup */
+        SIZE_T APInfo = sizeof(struct APInfo) + 2; 
+        PVOID PAPInfo = MmAllocateContiguousMemory(APInfo, HighestPhysicalAddress);
```

Comment:
Why is this not allocated from non-paged pool?

Reply:
It's the global per-processor state structure, see https://www.geoffchappell.com/studies/windows/km/ntoskrnl/inc/ntos/kpcr.htm ,
so "obviously" all that must be available at all times and therefore cannot be paged.

Reply:
That is why I asked "Why is this not allocated from **non-paged** pool?"

Reply:
Isn't the memory allocated with `MmAllocateContiguousMemory` in non-paged pool too?

Reply:
Nope, it takes space from the "system space" where e.g. drivers are mapped